src/main/java/com/synopsys/integration/synopsys/detect/scripts/scripts/ScriptBuilder.java

```diff
@@ -37,20 +37,27 @@
 import com.synopsys.integration.util.ResourceUtil;
 
 public class ScriptBuilder {
-    // To add a new Detect Major version (to generate scripts for), just increment getDetectMajorVersionLatest:
-    private final int getDetectMajorVersionLatest = 7;
+    // ScriptBuilder will generate scripts from the current templates for each Detect major version in this range:
+    // (There should be no breaking script changes across versions within this range.)
+    // When there is a breaking script change in either script, earlier versions (before the breaking change) of both scripts (not templates)
+    // must be preserved in src/main/resources/earlierversions.
+    private static final int DetectMajorVersionRangeEnd = 8;
+    private static final int DetectMajorVersionRangeStart = 8;
```

Comment:
```suggestion
    private static final int DETECT_MAJOR_VERSION_RANGE_START = 8;
```
I think this is the formatting we want

Reply:
Yeah, that doesn't make sense any more (if it ever did). I think that was the intent, but it only makes sense if we only make non-breaking changes. I've removed it.